recipes/proposal/all/conanfile.py

```diff
@@ -0,0 +1,101 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.32.0"
+
+
+class PROPOSALConan(ConanFile):
+    name = "proposal"
+    homepage = "https://github.com/tudo-astroparticlephysics/PROPOSAL"
+    license = "LGPL-3.0"
+    exports_sources = "CMakeLists.txt"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "monte Carlo based lepton and photon propagator"
+    topics = ("propagator", "lepton", "photon", "stochastic")
+
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_python": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_python": False,
+    }
+
+    generators = "cmake", "cmake_find_package"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("cubicinterpolation/0.1.4")
+        self.requires("spdlog/1.8.2")
+        if self.options.with_python:
+            self.requires("pybind11/2.6.2")
+
+    @property
+    def _minimum_compilers_version(self):
+        return {"gcc": "5", "clang": "6", "apple-clang": "6"}
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, "14")
+        if (
+            self.settings.os == "Windows"
+            or self.settings.compiler == "Visual Studio"
+        ):
+            raise ConanInvalidConfiguration(
+                "PROPOSAL is not supported on Windows and Visual Studio"
+            )
```

Comment:
Not supported or not tested? What's wrong with Visual Studio? What about MinGW?

Reply:
I've tested a little bit:
- MinGW probably works
- Visual Studio can't work due to usage of several non standard features:
  - massive usage of `or`, `not`, `and` keywords
  - several template declaration & definitions are not fully neat
  - several designated initializers (officially only supported in C++20)

Reply:
First of all thank you for your review!
Until now, PROPOSAL hasn't been developed to be compatible with Windows. The reason for that is that none of us developers work on Windows systems, therefore we never tested it. Furthermore, we only had very few people who asked for windows compatibility so far.

Of course we would be willing to adapt the code to be compatible for Windows, especially if the fixes are easy.

> several template declaration & definitions are not fully neat. Easy to fix.

Could you clarify this point, or give an example?

Reply:
The way `transform_loss` & `retransform_loss` specialization are defined 2 times in header and cpp file. It can't work with Visual Studio. Don't you want `extern` keyword?

Reply:
I'm able to build with Visual Studio with several patches. But for the moment `PROPOSAL::Helper::ResolvePath` is just a placeholder.
https://github.com/SpaceIm/conan-proposal/tree/e1daa5f36461bc63712a92f6b57314c9da101905/patches